tests/e2e/tests/simple/simple1_test.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Simple test - first time:
+source istio.VERSION
+bazel run //tests/e2e/tests/simple:go_default_test -- -alsologtostderr -test.v -v 2 \
+    -test.run TestSimple1 --skip_cleanup --auth_enable --namespace=e2e
+After which to Retest:
+bazel run //tests/e2e/tests/simple:go_default_test -- -alsologtostderr -test.v -v 2 \
+    -test.run TestSimple1 --skip_setup --skip_cleanup --auth_enable --namespace=e2e
+*/
+
+package simple
+
+import (
+	"flag"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+
+	"istio.io/istio/devel/fortio"
+	"istio.io/istio/tests/e2e/framework"
+	"istio.io/istio/tests/e2e/util"
+)
+
+const (
+	servicesYaml    = "tests/e2e/tests/simple/servicesToBeInjected.yaml"
+	nonInjectedYaml = "tests/e2e/tests/simple/servicesNotInjected.yaml"
+)
+
+type testConfig struct {
+	*framework.CommonConfig
+}
+
+var (
+	tc *testConfig
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	check(framework.InitGlog(), "cannot setup glog")
+	check(setTestConfig(), "could not create TestConfig")
+	os.Exit(tc.RunTest(m))
+}
+
+func TestSimpleIngress(t *testing.T) {
+	// Tests the rewrite/dropping of the /fortio/ prefix as fortio only replies
+	// with "echo debug server ..." on the /debug uri.
+	url := "http://" + tc.Kube.Ingress + "/fortio/debug"
+	glog.Infof("Fetching '%s'", url)
+	attempts := 7 // should not take more than 70s to be live...
```

Comment:
thought: we should really consider having some readiness check for Ingress.

Reply:
yes in 0.3 I hope we build up more of library functions, this is just a start using what we already have (it does add a handy getPodList() for instance)